api/db/services/doc_metadata_service.py

```
@@ -113,7 +112,7 @@ def _extract_metadata(flat_meta: Dict) -> Dict:
         return {}
 
     @staticmethod
-    def _extract_doc_id(doc: Dict, hit: Dict = None) -> str:
+    def _extract_doc_id(doc: dict, hit: dict = None) -> str:
```

Ruff (0.16.1)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)